File: js/signaldock/tests/integration/security-policy.test.js
const test = require('node:test');
const assert = require('node:assert/strict');
const { nextPolicy } = require('../../src/core/policy');
const { shouldShed } = require('../../src/core/queue');

test('policy and queue pressure align', () => {
  assert.equal(nextPolicy('normal', 2), 'watch');
  assert.equal(shouldShed(15, 10, false), true);
});

File: ruby/mercuryledger/tests/integration/security_policy_test.rb
# frozen_string_literal: true

require_relative '../test_helper'

class SecurityPolicyTest < Minitest::Test
  def test_policy_queue_alignment
    assert_equal 'watch', MercuryLedger::Core::Policy.next_policy('normal', 2)
    assert MercuryLedger::Core::Queue.should_shed?(15, 10, false)
  end
end
